californium-core/src/main/java/org/eclipse/californium/core/network/MessageExchangeStore.java

```diff
@@ -181,6 +184,15 @@
 	boolean isEmpty();
 
 	/**
+	 * Checks if there are any exchanges currently being registered in this
+	 * store and dump these exchanges. Intended to be used for more details in
+	 * log when testing.
+	 * 
+	 * @return {@code true} if there no exchanges registered.
+	 */
+	boolean isEmptyWithDump();
```

Comment:
For me this seems to be the classic "waste of time" discussion!

First, the main cause of this discussion seems to be, that the `MessageExchangeStore` / `InMemoryMessageExchangeStore` is stale but just not removed. In consequence, the discussion goes for "how should the new stuff be introduced?" Assuming, that this pattern (interface/implementation) is still valid or that it is already deprecated?

The arguments for placing the `isEmptyWithDump` are the same as for the already available `isEmpty`. So from my point of view, the task "adding more logging" MUST place the `isEmptyWithDump` in the same way then `isEmpty`. 
Redesign `MessageExchangeStore` / `InMemoryMessageExchangeStore` is a different task and may change that placing again.